frame/pablo/src/stable_swap.rs

```diff
@@ -79,7 +79,10 @@ impl<T: Config> StableSwap<T> {
 		let pair = if asset_id == pool.pair.base { pool.pair } else { pool.pair.swap() };
 		let pool_base_aum = T::Assets::balance(pair.base, pool_account);
 		let pool_quote_aum = T::Assets::balance(pair.quote, pool_account);
-		ensure!(!pool_base_aum.is_zero() && !pool_quote_aum.is_zero(), Error::<T>::NotEnoughLiquidity);
+		ensure!(
```

Comment:
The problem was that when one calls the `pablo_priceFor` rpc, this was causing an `arithmatic overflow`, when there is no liquidity in the pool initially. So it's not a rare case. Alternative solutions are welcome though :)

Reply:
> that is solution - find overflow case and map_err into liquidity error.

In the RPC there is no way to know what causes an `overflow`, so can not reliably map to liquidity error, hence I decided to map it here. When there is UI integration for pools, calling this RPC with 0 liquidity can be common as one checks out available pools. 

> non rare case is swap of amount larger than pool, not zero pool.

Not sure about this case as that is related to user input as opposed to a system error. May be you can clarify more?

Reply:
Perhaps instead of checking for zero , we can check if pool can process given value in exchange ? That way all cases will be handled and that fills true case of NotEnoughLiquidity

Reply:
I think it's fine assuming a pool should own at least 10^-12 unit of each asset :)

Reply:
for me this code is not needed. we just burn gas for no valid reason.

Reply:
@dzmitry-lahoda I'm not sure if this amount of gas is that significant, I will revert the change if this is seen as costly in the future :)

Reply:
> we can check if pool can process given value in exchange ?
@vivekvpandya This is exactly the function that is being used for that

Reply:
@vimukthi-git  what I meant is like say a user wants 10 Base_coin and pool only has 9 Base_coin in that case also it is not_enough_liquidity right? or if user has not specified min_receive then pool will give user 9 base_coin?

Reply:
Hmm that is a separate case entirely as that does not cause any overflow errors.